src/_pytest/pytester.py

```diff
@@ -534,7 +534,8 @@ class Testdir:
 
     __test__ = False
 
-    CLOSE_STDIN = object
+    class CLOSE_STDIN:
+        pass
```

Comment:
```suggestion
        __slots__ = ()
```

Reply:
Hmm given that you use the class itself as the sentinel and not an instance of it, this suggestion doesn't matter much...

Reply:
an enum instance is usually pretty good

```python3
class CloseStdinType(enum.Enum):
    CLOST_STDIN = enum.auto()
```

it's great for type annotations too

Reply:
and it's got that None/NoneType feel

Reply:
@blueyed you can do stuff like:
```
def popen(self, ..., stdin=CLOSE_STDIN: Union[CloseStdinType,bytes,int]): -> ...
```

Reply:
@graingert Right, this is the best current way to do it in type-compatible way: https://github.com/python/typing/issues/689. Also what I opted for here: https://github.com/bluetech/pytest/commit/bcd19fac5b30882e106ef1e3dbad4bb7edb8b303#diff-4ec37b1ad5b7d41fdac56c907d08dde1R46-R52

Reply:
What's wrong with `stdin: Optional[Union[IO[str], "EncodedFile", "DontReadFromInput", Type[CLOSE_STDIN]]] = CLOSE_STDIN,`?
It might be not as strict really, but serves the purpose in general, and has less/no boilerplate.

Reply:
Nothing wrong really, just that using a `Type[...]` as a sentinel is odd/surprising. I'd go with @graingert's suggestion, which is the ["official"](https://www.python.org/dev/peps/pep-0484/#support-for-singleton-types-in-unions) way to do it.

That said if you prefer the Type that's fine.